satpy/readers/yaml_reader.py

```diff
@@ -493,6 +494,52 @@ def create_filehandlers(self, filenames):
                     filehandlers,
                     key=lambda fhd: (fhd.start_time, fhd.filename))
 
+        # update existing dataset IDs with information from the file handler
+        self.update_ds_info_from_files()
+
+        # load any additional dataset IDs determined dynamically from the file
+        self.update_ds_ids_from_files()
+
+    def update_ds_info_from_files(self):
+        """Update DatasetIDs with information from loaded files.
+
+        This is useful, for example, if dataset resolution may change
+        depending on what files were loaded.
+
+        """
+        for file_handlers in self.file_handlers.values():
+            fh = file_handlers[0]
+            # update resolution in the dataset IDs for this files resolution
+            try:
+                res = fh.resolution
+            except NotImplementedError:
+                continue
+
+            for ds_id, ds_info in list(self.ids.items()):
+                if fh.filetype_info['file_type'] != ds_info['file_type']:
+                    continue
+                if ds_id.resolution is not None:
+                    continue
+                ds_info['resolution'] = res
+                new_id = DatasetID.from_dict(ds_info)
+                self.ids[new_id] = ds_info
+                del self.ids[ds_id]
```

Comment:
Are we sure this doesn't mess up any dependency tree ?

Reply:
This happens before the dependency tree is created/loaded. This is updating existing IDs from the YAML with information from the available files. This all happens in the "here are my files, analyze them" step.

Reply:
ok, good.